src/features/rbac/server/projectMembersRouter.ts

```diff
@@ -91,14 +124,39 @@ export const projectMembersRouter = createTRPCRouter({
           email: input.email.toLowerCase(),
         },
       });
-      if (!user) throw new Error("User not found");
+      if (user) {
+        return await ctx.prisma.membership.create({
+          data: {
+            userId: user.id,
+            projectId: input.projectId,
+            role: input.role,
+          },
+        });
+      }
 
-      return await ctx.prisma.membership.create({
+      const invitation = await ctx.prisma.projectInvitation.create({
         data: {
-          userId: user.id,
           projectId: input.projectId,
+          email: input.email.toLowerCase(),
           role: input.role,
+          senderId: ctx.session.user.id,
+        },
+      });
+
+      const project = await ctx.prisma.project.findFirstOrThrow({
+        where: {
+          id: input.projectId,
         },
       });
+
+      if (!project) throw new Error("Project not found");
```

Comment:
When using findFirstOrThrow, it already throws if none is found. Better to use findFirst and throw error yourself like you did